src/controllers/oracle.controller.ts

```diff
@@ -0,0 +1,287 @@
+import {HttpErrors, operation, param, requestBody} from '@loopback/rest';
+import {GoodOracle, GoodOracleInput} from '../models';
+import {GoodOracleRepository} from '../repositories';
+
+export class OracleController {
+  constructor(
+    @repository(GoodOracleRepository)
+    public goodOracleRepository: GoodOracleRepository,
+  ) {}
+
+  /**
+   * Create a new Oracle
+   *
+   * @param id The PoG ID of the oracle
+   * @param oracle
+   */
+  @operation('post', '/oracle/{id}', {
+    summary: 'Create an Oracle',
+    operationId: 'post-oracle',
+    responses: {
+      '201': {
+        description: 'Oracle Created',
+      },
+      '400': {
+        description: 'Missing Required Information',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+      '401': {
+        description: 'Unauthorized',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+      '404': {
+        description: 'Oracle Not Found',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+    },
+    requestBody: {
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodOracle_Input',
+          },
+          examples: {
+            'Create an Oracle': {
+              value: {
+                name: 'Leyline',
+                goodOracleURI: 'leyline.gg',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    },
+    description: 'Create a new Oracle',
+    parameters: [
+      {
+        schema: {
+          type: 'integer',
+          example: 3,
+        },
+        name: 'id',
+        in: 'path',
+        required: true,
+        description: 'The PoG ID of the oracle',
+      },
+    ],
+    security: [
+      {
+        Oracle_API_Key: [],
+      },
+    ],
+  })
+  async postOracle(
+    @param({
+      schema: {
+        type: 'integer',
+        example: 3,
+      },
+      name: 'id',
+      in: 'path',
+      required: true,
+      description: 'The PoG ID of the oracle',
+    })
+    id: number,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodOracle_Input',
+          },
+          examples: {
+            'Create an Oracle': {
+              value: {
+                name: 'Leyline',
+                goodOracleURI: 'leyline.gg',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    })
+    oracle: GoodOracle,
+  ): Promise<unknown> {
+    oracle.id = id;
+    return this.goodOracleRepository.create(oracle);
+  }
```

Comment:
I'm finding it weird to have a create with an id that is required.
Shouldn't we obtain that id after creating it ?
the post should be a simple `POST /oracle`, shouldn't it?

Reply:
yes, I think it should be, good point